### PR TITLE
Improved Register System + cleanup

### PR DESCRIPTION
**Pull request by Mooy1**
## Changes
- Added back launch messages
- Changed CelestialWorld to PlanetaryObject
- Changed register methods for Alien and PlanetaryObject
- Changed how Aliens are added to AlienWorlds
- Cleaned up Utils
- Updated InfinityLib
- Added StructureManager
- Moved some stuff to constructor in Alien and most UniversalObjects
- Improved BossBarStyle
- Moved WorldSetting stuff to methods in WorldManager and AlienWorld
- Changed Orbit to use radians
- Updated listeners to private and added priorities
- Changed fuel system to stored fuel type rather than efficiency cuz mixing fuels is probably bad
- Removed ItemChoice

## Related Issues
x

## Checklist

- [ ] I have fully tested the proposed changes and promise that they will not break world generation, mob spawning, and the like.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

### CODE REVIEW

File: src/main/java/io/github/addoncommunity/galactifun/api/worlds/WorldManager.java [hunk 1]
```diff
@@ -1,109 +1,112 @@
...
+                this.config.load(configFile);
+            } catch (Exception e) {
                 e.printStackTrace();
             }
-        } catch (IOException e) {
-            e.printStackTrace();
         }
+
+        // Save the config after startup
+        galactifun.runAsync(() -> {
```
Comment (Seggan) [inline src/main/java/io/github/addoncommunity/galactifun/api/worlds/WorldManager.java:69]: bukkit methods arent thread safe

File: src/main/java/io/github/addoncommunity/galactifun/api/worlds/WorldManager.java [hunk 2]
```diff
@@ -138,8 +141,8 @@ public void onPlayerTeleport(@Nonnull PlayerTeleportEvent e) {
         }
     }
 
-    @EventHandler
-    public void onCreatureSpawn(@Nonnull CreatureSpawnEvent e) {
+    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
```
Comment (Seggan) [inline src/main/java/io/github/addoncommunity/galactifun/api/worlds/WorldManager.java:144]: you know highest = first executed, right?

File: src/main/java/io/github/addoncommunity/galactifun/base/universe/saturn/Titan.java [hunk 3]
```diff
@@ -56,20 +56,14 @@ protected void generateChunk(@Nonnull ChunkGenerator.ChunkData chunk, @Nonnull C
...
-            for (z = 0, realZ = chunkZ << 4; z < 16; z++, realZ++) {
+        for (int x = 0, realX = chunkX << 4; x < 16; x++, realX++) {
+            for (int z = 0, realZ = chunkZ << 4; z < 16; z++, realZ++) {
 
                 chunk.setBlock(x, 0, z, Material.BEDROCK);
 
                 // find max height
                 double startHeight = generator.noise(realX, realZ, 0.5, 0.5, true);
-                height = (int) (55 + 30 * + (startHeight * startHeight));
+                int height = (int) (55 + 30 * + (startHeight * startHeight));
```
Comment (Seggan) [inline src/main/java/io/github/addoncommunity/galactifun/base/universe/saturn/Titan.java:66]: erm `30 * +`????

File: src/main/java/io/github/addoncommunity/galactifun/core/CoreCategory.java [hunk 4]
```diff
@@ -45,7 +45,7 @@
 
     public static void setup(Galactifun galactifun) {
         Category universe = new GalacticCategory(galactifun.getKey("galactic_flex"),
-                new CustomItem(Material.END_STONE, "&bThe Universe"), galactifun.getGalacticExplorer());
+            new CustomItem(Material.END_STONE, "&bThe Universe"), new GalacticExplorer()); // TODO explorer could be used in rockey
```
Comment (Seggan) [inline src/main/java/io/github/addoncommunity/galactifun/core/CoreCategory.java:48]: rocket

File: src/main/java/io/github/addoncommunity/galactifun/core/commands/SphereCommand.java [hunk 5]
```diff
@@ -58,13 +57,8 @@ public void onExecute(@Nonnull CommandSender commandSender, @Nonnull String[] st
...
         
-        Sphere sphere = new Sphere(materials);
-        
-        double time = System.nanoTime();
-        
-        sphere.generate(target, radius, 0);
-        
-        p.sendMessage(ChatColor.GREEN + "Generated in " + Util.timeSince(time));
+        new Sphere(materials).generate(target, radius, 0);
+        p.sendMessage(ChatColor.GREEN + "Generated!");
```
Comment (Seggan) [inline src/main/java/io/github/addoncommunity/galactifun/core/commands/SphereCommand.java:61]: timings still might help

File: src/main/resources/config.yml [hunk 6]
```diff
@@ -9,13 +9,21 @@ worlds:
...
 
   # maximum number of aliens per player per world
   max-per-player: 8
 
 rockets:
-
-  # the messages that pop up during launch, commented out because of infinitylib problems
-  #launch-msgs: [ Forgetting parachutes, Restocking snack bin, Aligning planets, Igniting sun, Inventing funny launch messages, Thanking TheBusyBiscuit, Faking Moon landings, Taking one small step ]
+  
+  # Messages while rocket it launching
```
Comment (Seggan) [inline src/main/resources/config.yml:20]: is

**Review by Seggan — CHANGES_REQUESTED** [review]
Lotsa typos

File: src/main/java/io/github/addoncommunity/galactifun/api/universe/types/PlanetaryType.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
+package io.github.addoncommunity.galactifun.api.universe.types;
+
+/**
+ * Type of celestial body
+ * 
+ * @author Mooy1
+ */
+public final class PlanetaryType extends UniversalType {
```
[the remark this answers is not in the document]
Comment (Mooy1) [reply src/main/java/io/github/addoncommunity/galactifun/api/universe/types/PlanetaryType.java:8]: ye i figured some people may want to make their own

File: src/main/java/io/github/addoncommunity/galactifun/base/universe/saturn/Titan.java (hunk 3 above)
Comment (Mooy1) [reply src/main/java/io/github/addoncommunity/galactifun/base/universe/saturn/Titan.java:66]: dunno, thats how it was before lol

File: src/main/java/io/github/addoncommunity/galactifun/api/worlds/WorldManager.java (hunk 2 above)
Comment (Mooy1) [reply src/main/java/io/github/addoncommunity/galactifun/api/worlds/WorldManager.java:144]: it goes low -> high then monitor

File: src/main/java/io/github/addoncommunity/galactifun/api/aliens/Alien.java [hunk 8]
```diff
@@ -96,7 +109,7 @@ protected void onMobTick(@Nonnull Mob mob) { }
     public final int attemptSpawn(Random rand, World world) {
         int spawned = 0;
         for (Chunk chunk : world.getLoadedChunks()) {
-            if (rand.nextInt(100) > getSpawnChance() || spawned >= getMaxAliensPerGroup()) {
+            if (rand.nextInt(100) > this.spawnChance) {
```
[the remark this answers is not in the document]
Comment (Mooy1) [reply src/main/java/io/github/addoncommunity/galactifun/api/aliens/Alien.java:112]: no random.nextDouble(bound)

**Comment by Seggan** [comment]
Gonna review when I have time

**Review by Seggan — APPROVED** [review]

**Comment by Mooy1** [comment]
Should probably be merged before any other request because i can't cleanup any conflicts for 2 weeks

**Comment by Seggan** [comment]
I'm always leave the owner to merge

**Pull request merged**